### PR TITLE
Add shorthand alternatives for command line options in OSB

### LINKED ISSUE #639: Add short-hand alternatives for commonly-used command line options
### Is your feature request related to a problem? Please describe

OSB has a few command-line options that offer short hand alternatives. An example is the following, added by @peteralfonsi .
```
    test_execution_parser.add_argument(
        "--randomization-repeat-frequency",
        "-rf",
        help=f"The repeat_frequency for query randomization. Ignored if randomization is off"
             f"(default: {workload.loader.QueryRandomizerWorkloadProcessor.DEFAULT_RF}).",
        default=workload.loader.QueryRandomizerWorkloadProcessor.DEFAULT_RF)
```

This improves the user experience and removes the dread of typing out long command-line options.

### Describe the solution you'd like

We should include short hand alternatives for the following commonly-used options for each subcommand:

### execute-test 
- [ ] `--workload`
- [ ] `--target-hosts`
- [ ] `--client-options`
- [ ] `--workload-params`
- [ ] `--test-execution-id`
- [ ] `--kill-running-processes`

### compare

- [ ] `--baseline`
- [ ] `--contender`

### create-workload

- [ ] `--workload`
- [ ] `--indices`
- [ ] `--target-hosts`
- [ ] `--client-options`

By only focusing on the commonly-used options, we will avoid polluting OSB's namespace with convoluted short flags while also improving the user experience. If an option that's added in the future becomes commonly-used, we can add a short flag then.

If there are other commonly-used options that I missed, please feel free to comment them below.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by OVI3D0**
### Description
Adds short-hand alternatives for commonly used command line options in OSB.

Now, rather than running:
`opensearch-benchmark execute-test --workload=geonames --target-hosts=127.0.0.1:9200 --test-mode --kill-running-processes`

one can instead run:
`opensearch-benchmark execute-test -w=geonames -th=127.0.0.1:9200 --test-mode -krp`

### Issues Resolved
#639 

### Testing
Tested by using shorthand alternatives for the commands changed.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by IanHoang** [comment]
This will streamline the testing experience. Thanks for doing this!

**Review by IanHoang — APPROVED** [review]

**Comment by OVI3D0** [comment]
> Could we also add one for `opensearch-benchmark --version` (to match how `--help` has `-h`)?

Done!

File: osbenchmark/benchmark.py [hunk 1]
```diff
@@ -205,6 +213,7 @@ def add_workload_source(subparser):
         default=metrics.GlobalStatsCalculator.DEFAULT_LATENCY_PERCENTILES)
     compare_parser.add_argument(
         "--results-format",
+        "-rfo",
```
Comment (IanHoang) [inline osbenchmark/benchmark.py:216]: For results format and results file, I think we can just stick with the longer names and not assign them short-hand flags.

**Review by IanHoang — COMMENTED** [review]
Spoke offline with @OVI3D0: Will just include short hand flags for most common commands. Updated issue to reflect the most common commands only: https://github.com/opensearch-project/opensearch-benchmark/issues/639

File: osbenchmark/benchmark.py [hunk 2]
```diff
@@ -157,24 +158,29 @@ def add_workload_source(subparser):
...
         help="Name of the generated workload")
     create_workload_parser.add_argument(
         "--indices",
+        "-i",
         type=non_empty_list,
         required=True,
         help="Comma-separated list of indices to include in the workload")
     create_workload_parser.add_argument(
         "--target-hosts",
+        "-th",
```
[the remark this answers is not in the document]
Comment (OVI3D0) [reply osbenchmark/benchmark.py]: Done 👍

**Pull request merged**